src/main/java/com/rednavis/vaadin/showcase/backend/service/password/PasswordServiceImpl.java

```diff
@@ -0,0 +1,287 @@
+package com.rednavis.vaadin.showcase.backend.service.password;
+
+import com.rednavis.vaadin.showcase.backend.config.PasswordConfig;
+import edu.vt.middleware.password.AlphabeticalCharacterRule;
+import edu.vt.middleware.password.AlphabeticalSequenceRule;
+import edu.vt.middleware.password.DigitCharacterRule;
+import edu.vt.middleware.password.LengthRule;
+import edu.vt.middleware.password.LowercaseCharacterRule;
+import edu.vt.middleware.password.NonAlphanumericCharacterRule;
+import edu.vt.middleware.password.NumericalSequenceRule;
+import edu.vt.middleware.password.Password;
+import edu.vt.middleware.password.PasswordData;
+import edu.vt.middleware.password.PasswordValidator;
+import edu.vt.middleware.password.QwertySequenceRule;
+import edu.vt.middleware.password.RepeatCharacterRegexRule;
+import edu.vt.middleware.password.Rule;
+import edu.vt.middleware.password.RuleResult;
+import edu.vt.middleware.password.UppercaseCharacterRule;
+import edu.vt.middleware.password.WhitespaceRule;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.spec.InvalidKeySpecException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class PasswordServiceImpl implements PasswordService {
+
+  private static final String PBKDF2_ALGORITHM = "PBKDF2WithHmacSHA1";
+  // These constants may be changed without breaking existing hashes.
+  private static final int SALT_BYTE_SIZE = 24;
+  private static final int HASH_BYTE_SIZE = 18;
+  private static final int PBKDF2_ITERATIONS = 64000;
+  // These constants define the encoding and may not be changed.
+  private static final int HASH_SECTIONS = 5;
+  private static final int HASH_ALGORITHM_INDEX = 0;
+  private static final int ITERATION_INDEX = 1;
+  private static final int HASH_SIZE_INDEX = 2;
+  private static final int SALT_INDEX = 3;
+  private static final int PBKDF2_INDEX = 4;
+
+  private final PasswordValidator passwordValidator;
+
+  @Inject
+  public PasswordServiceImpl(PasswordConfig passwordConfig) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The constructor 'PasswordServiceImpl(PasswordConfig)' has an NPath complexity of 1024](https://app.codacy.com/gh/rednavis/vaadin-showcase/pullRequest?prid=4839326)